File: src/main/scala/com/rawlabs/das/jira/DASJiraBuilder.scala
/*
 * Copyright 2024 RAW Labs S.A.
 *
 * Use of this software is governed by the Business Source License
 * included in the file licenses/BSL.txt.
 *
 * As of the Change Date specified in that file, in accordance with
 * the Business Source License, use of this software will be governed
 * by the Apache License, Version 2.0, included in the file
 * licenses/APL.txt.
 */

package com.rawlabs.das.jira

import com.rawlabs.das.sdk.{DASSdk, DASSdkBuilder}
import com.rawlabs.utils.core.RawSettings

class DASJiraBuilder extends DASSdkBuilder {

  override def dasType: String = "jira"

  override def build(options: Map[String, String])(implicit settings: RawSettings): DASSdk = new DASJira(options)

}

File: das-jira-connector/src/main/java/com/rawlabs/das/jira/DASJiraBuilder.java
package com.rawlabs.das.jira;

import com.rawlabs.das.sdk.DASSdk;
import com.rawlabs.das.sdk.DASSdkBuilder;
import com.rawlabs.das.sdk.DASSettings;
import java.util.Map;

public class DASJiraBuilder implements DASSdkBuilder {

  @Override
  public String getDasType() {
    return "jira";
  }

  public DASSdk build(Map<String, String> options, DASSettings rawSettings) {
    return new DASJira(options);
  }
}
